Lib/test/test_binascii.py

```
@@ -159,6 +162,9 @@ def assertDiscontinuousPadding(data, non_strict_mode_expected_result: bytes):
         assertLeadingPadding(b'===', b'')
         assertDiscontinuousPadding(b'ab=c=', b'i\xb7')
         assertDiscontinuousPadding(b'ab=ab==', b'i\xb6\x9b')
+        assertIncorrectPadding(b'AAAA=', b'\x00\x00\x00')
```

For consistency with other tests it is better to use `abcd` instead of `AAAA`. It also allows to see what data was ignored.

Please add also tests for 4 and more padding `=`s: `b'===='` and `b'abcd===='`, `b'====='` and `b'abcd====='`.

Also tests for extra `=` after correct padding: `b'abc=='`,  `b'abc==='`,  `b'abc==='`,  `b'abc===='`,  `b'abc====='`,  `b'ab==='`,  `b'ab===='`, etc.

---

@serhiy-storchaka Cases like `b'ab==='` is reported as `Excess data after padding` rather than `Excess padding`. Is this behavior OK for you?

---

I have no preferences. What is important is that it raises an error instead of silently accepting.